### ISSUE TITLE
Fix and enable ValhallaAttributeTests variations 4/5/6

### ISSUE THREAD

**Issue by theresa-m** [eclipse-openj9/openj9#18742]
https://github.com/eclipse-openj9/openj9/pull/18597 enables flattening for nullrestricted fields. This new behavior shows that a null check is missing for nullrestricted field assignment in the flattened field case.

See https://github.com/eclipse-openj9/openj9/pull/18597#issuecomment-1887485337 for more information

This is the failing test build but it should be straight forward to reproduce locally https://openj9-jenkins.osuosl.org/job/Test_openjdkValhalla_j9_sanity.functional_x86-64_windows_valhalla_Personal/23/

FYI @a7ehuo

**Comment by hangshao0** [comment]
The build in the description could be gone after a period of time, I am pasting the failure here:

```
19:43:25  FAILED: testPutFieldNullToNullRestrictedField
19:43:25  org.testng.TestException: 
19:43:25  Method ValhallaAttributeTests.testPutFieldNullToNullRestrictedField()[pri:2, instance:org.openj9.test.lworld.ValhallaAttributeTests@213f8378] should have thrown an exception of type class java.lang.NullPointerException
19:43:25  	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:80)
19:43:25  	at org.testng.internal.Invoker.considerExceptions(Invoker.java:1139)
19:43:25  	at org.testng.internal.Invoker.invokeMethod(Invoker.java:612)
19:43:25  	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716)
19:43:25  	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988)
19:43:25  	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
19:43:25  	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
19:43:25  	at org.testng.TestRunner.privateRun(TestRunner.java:648)
19:43:25  	at org.testng.TestRunner.run(TestRunner.java:505)
19:43:25  	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
19:43:25  	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
19:43:25  	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
19:43:25  	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
19:43:25  	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
19:43:25  	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
19:43:25  	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
19:43:25  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
19:43:25  	at org.testng.TestNG.runSuites(TestNG.java:1049)
19:43:25  	at org.testng.TestNG.run(TestNG.java:1017)
19:43:25  	at org.testng.TestNG.privateMain(TestNG.java:1354)
19:43:25  	at org.testng.TestNG.main(TestNG.java:1323)
```

**Comment by a7ehuo** [comment]
I can reproduce the failure in `testPutFieldNullToNullRestrictedField`  locally. I'll take a look at it next

@hzongaro fyi

**Comment by a7ehuo** [comment]
The root cause of the issue is that when `TestPutFieldNullToNullRestrictedFieldField` doesn't have any field and it is flattened into `TestPutFieldNullToNullRestrictedField`,  `TestPutFieldNullToNullRestrictedField` has zero field count in [TR::TypeLayout](https://github.com/eclipse-openj9/openj9/blob/ed3e66243a653b4b0b37c48c52c26dde9ab68397/runtime/compiler/env/J9ClassEnv.cpp#L544-L546). The [storeInstance](https://github.com/eclipse-openj9/openj9/blob/ed3e66243a653b4b0b37c48c52c26dde9ab68397/runtime/compiler/ilgen/Walker.cpp#L7210) code doesn't take into consider of this case. This issue is being fixed in #18754.


```java
public value class TestPutFieldNullToNullRestrictedFieldField {
   // zero field
}
 
public class TestPutFieldNullToNullRestrictedField {

   TestPutFieldNullToNullRestrictedFieldField field;  
}
```

**Comment by hangshao0** [comment]
@a7ehuo @hzongaro Thanks for fixing this !

The excluded tests have been re-enabled by https://github.com/eclipse-openj9/openj9/pull/18754.
